transformations.py

```diff
@@ -0,0 +1,114 @@
+import pandas as pd
+import re
+import time
+
+
+class Field_Transformations:
+    def __init__(self, dataframe):
+        self.dataframe = dataframe
+        self.possition_abbr()
+        self.add_dept_codes()
+        self.parse_pay_location()
+        self.parse_work_location()
+        self.clean_address()
+        self.clean_homePhone()
+        self.clean_hireDate()
+
+    """
+    When paycom_job_title is formatted as "Academics Associate (0002/1005/ADASO)"", the following 
+    method extracts the capitalized, abbreviated form of the position (i.e. ADASO). Not all field values are
+    in this format so it first checks if "(" is included in the string
+    """
+
+    def possition_abbr(self):
+        new_column = []
+        for result in self.dataframe["paycom_job_title"]:
+            if "(" in result:
+                value = result.split("/")[2]
+                new_column.append(value[:-1])
+            else:
+                new_column.append(None)
+        self.dataframe["position"] = new_column
+
+    """
+    The following method first joins our Jobvite dataframe with the csv containing department codes, then it cleans it 
+    by turning the float (ex. "1015.0") into a string and removing ".0" by only taking the first 4 characters ([:4])
+    """
+
+    def add_dept_codes(self):
+        dept_codes = pd.read_csv("dept_codes.csv")
+        self.dataframe = pd.merge(
+            self.dataframe, dept_codes, on="department", how="left"
+        )
+        self.dataframe.drop(columns="Unnamed: 0", inplace=True)
+        new_dept_column = []
+        for result in self.dataframe["dept_code"]:
+            if isinstance(result, float):
+                new_value = str(result)[:4]
+                new_dept_column.append(new_value)
+            else:
+                new_dept_column.append(None)
+
+        self.dataframe["dept_code"] = new_dept_column
+
+    ##Paycom only accepts words and digits within address column so this method removes unacceptable characters
+    def clean_address(self):
+        new_addr_column = []
+        for result in self.dataframe["address"]:
+            new_value = (
```

Comment:
Again, I'd avoid the for loop here for modifying the data frame, but a for loop is actually pretty useful for making the multiple replaces:

```python
special_characters = ".#,/" 
for character in special_characters:
    result = result.replace(character, "")
```